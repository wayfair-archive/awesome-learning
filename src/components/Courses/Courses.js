import React, {useState} from 'react';
import PropTypes from 'prop-types';
import {makeStyles, useTheme} from '@material-ui/core/styles';
import {Box, Typography, Button} from '@material-ui/core';
import {Pagination} from '@material-ui/lab';
import {Link} from 'gatsby';

const ITEMS_PER_PAGE = 4;

const useStyles = makeStyles((theme) => ({
  backToContainer: {
    marginBottom: theme.spacing(4),
  },
  title: {
    width: 'fit-content',
    margin: theme.spacing(1, 0, 2),
    textDecoration: 'underline',
    textTransform: 'none',
    '&:hover, &:focus': {
      textDecoration: 'none',
    },
  },
  courseContainer: {
    margin: theme.spacing(5, 0, 14),
  },
  goLearnCTA: {
    width: '140px',
    marginTop: theme.spacing(4),
  },
}));

const Courses = ({edges, title}) => {
  const classes = useStyles();
  const theme = useTheme();
  const [page, setPage] = useState(1);
  const noOfPages = Math.ceil(edges.length / ITEMS_PER_PAGE);
  const handleChange = (event, value) => {
    setPage(value);
    window.scrollTo(0, 0);
  };

  return (
    <Box m="auto" maxWidth={theme.breakpoints.values.lg}>
      <Typography variant="h1" color="textPrimary">
        {title}
      </Typography>
      {edges
        .slice((page - 1) * ITEMS_PER_PAGE, page * ITEMS_PER_PAGE)
        .map((edge) => (
          <Box
            display="flex"
            flexDirection="column"
            key={edge.node.frontmatter.title}
            className={classes.courseContainer}
          >
            <Typography
              variant="h2"
              component={Link}
              color="primary"
              to={edge.node.fields.slug}
              className={classes.title}
            >
              {edge.node.frontmatter.title}
            </Typography>
            <Typography variant="body1">
              {edge.node.frontmatter.description}
            </Typography>
            <Button
              variant="contained"
              color="primary"
              component={Link}
              to={edge.node.fields.slug}
              className={classes.goLearnCTA}
            >
              Go Learn
            </Button>
          </Box>
        ))}
      <Box display="flex" justifyContent="center">
        <Pagination
          count={noOfPages}
          page={page}
          onChange={handleChange}
          defaultPage={1}
          variant="outlined"
          shape="rounded"
          size="large"
        />
      </Box>
    </Box>
  );
};

Courses.propTypes = {
  edges: PropTypes.arrayOf(PropTypes.object).isRequired,
  title: PropTypes.string,
};

Courses.defaultProps = {
  title: '',
};

export default Courses;
