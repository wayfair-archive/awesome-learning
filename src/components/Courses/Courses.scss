.CourseList {
  $rootSelector: &;
  &-item {
    @include margin-bottom(1.25);
    &:last-child {
      @include margin-bottom(0.5);
    }
  }

  &-card {
    display: flex;
    flex-direction: column;
    align-items: center;
    text-align: center;
    background-color: #fff;
    text-decoration: none;
    padding: $spacing--medium;
    height: 100%;
    border: 1px solid #00a7ba;
  }

  &-itemTitle {
    @include margin-bottom(0.5);
  }

  &-itemDescription {
    @include line-height(1);
    @include margin-bottom(0.75);
  }

  &--isHorizontal {
    @include margin-top(2);
    display: flex;
    flex-flow: row wrap;

    @include breakpoint-xs {
      justify-content: center;
      text-align: center;
      #{$rootSelector}-item {
        width: 80%;
      }
    }

    @include breakpoint-sm {
      #{$rootSelector}-item {
        width: 30%;
      }
    }
  }
}
