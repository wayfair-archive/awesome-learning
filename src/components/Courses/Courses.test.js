import React from 'react';
import {render} from '@testing-library/react';
import Courses from './Courses';

const PROPS = {
  edges: [
    {
      node: {
        fields: {
          slug: '/courses/Accessibility/',
          categorySlug: '/category/accessibility/',
        },
        frontmatter: {
          title: 'Accessibility',
          category: 'Accessibility',
          description:
            'Accessibility refers to how people with disabilities can meaningfully use products...',
        },
      },
    },
    {
      node: {
        fields: {
          slug: '/courses/Array-Methods/',
          categorySlug: '/category/js-fundamentals/',
        },
        frontmatter: {
          title: 'Array Methods',
          category: 'JS Fundamentals',
          description:
            'Welcome to the Awesome Learning Array Methods course. Over the next 5 lessons, ...',
        },
      },
    },
    {
      node: {
        fields: {
          slug: '/courses/CSS-Layouts/',
          categorySlug: '/category/css-fundamentals/',
        },
        frontmatter: {
          title: 'CSS Layouts',
          category: 'CSS Fundamentals',
          description:
            'Need help achieving the ideal layout for your feature? This course explains the...',
        },
      },
    },
    {
      node: {
        fields: {
          slug: '/courses/Data-Types/',
          categorySlug: '/category/js-fundamentals/',
        },
        frontmatter: {
          title: 'Data Types',
          category: 'JS Fundamentals',
          description:
            'The focus of this course is to make you familiar with the nuance of the JavaScript...',
        },
      },
    },
    {
      node: {
        fields: {
          slug: '/courses/Functions-and-Scope/',
          categorySlug: '/category/js-fundamentals/',
        },
        frontmatter: {
          title: 'Functions and Scope',
          category: 'JS Fundamentals',
          description:
            'If you want to learn about JavaScript functions, then this is the course for you...',
        },
      },
    },
    {
      node: {
        fields: {
          slug: '/courses/Promises/',
          categorySlug: '/category/async-js/',
        },
        frontmatter: {
          title: 'Promises',
          category: 'Async JS',
          description:
            'Asynchronous programming patterns are the cornerstone of Frontend. All of the...',
        },
      },
    },
  ],
};

describe('Courses', () => {
  it('renders only the first four courses because of pagination', () => {
    const {queryByText, queryAllByText} = render(<Courses {...PROPS} />);

    expect(queryAllByText('Accessibility')).toBeTruthy();
    expect(queryByText('Array Methods')).toBeTruthy();
    expect(queryByText('CSS Layouts')).toBeTruthy();
    expect(queryByText('Data Types')).toBeTruthy();
    expect(queryByText('Functions and Scope')).toBeFalsy();
    expect(queryByText('Promises')).toBeFalsy();
  });

  it('renders pagination bar', () => {
    const {queryByLabelText} = render(<Courses {...PROPS} />);
    expect(queryByLabelText('Go to next page')).toBeTruthy();
  });
});
