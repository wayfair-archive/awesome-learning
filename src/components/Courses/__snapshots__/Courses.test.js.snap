// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feed renders correctly 1`] = `
<div
  className="CourseList"
>
  <div
    className="CourseList-item"
  >
    <Link
      activeClassName=""
      className="Link Link--tertiary"
      to="/test_0"
    >
      test_0
    </Link>
    <div
      className="CourseList-itemTitle"
    >
      <Link
        activeClassName=""
        className="Link Link--primary"
        to="/test_0"
      >
        test_0
      </Link>
    </div>
    <p
      className="CourseList-itemDescription"
    >
      test_0
    </p>
    <Link
      activeClassName=""
      className="Link Link--pill"
      to="/test_0"
    >
      Go Learn
    </Link>
  </div>
  <div
    className="CourseList-item"
  >
    <Link
      activeClassName=""
      className="Link Link--tertiary"
      to="/test_1"
    >
      test_1
    </Link>
    <div
      className="CourseList-itemTitle"
    >
      <Link
        activeClassName=""
        className="Link Link--primary"
        to="/test_1"
      >
        test_1
      </Link>
    </div>
    <p
      className="CourseList-itemDescription"
    >
      test_1
    </p>
    <Link
      activeClassName=""
      className="Link Link--pill"
      to="/test_1"
    >
      Go Learn
    </Link>
  </div>
</div>
`;
